include/shell.h

```diff
@@ -55,7 +55,7 @@ class ByteReader {
 
 class HostShell {
 public:
-	virtual bool GetEnvStr(const char* entry, std::string& result) const = 0;
+	virtual std::optional<std::string> GetEnvStr(std::string_view entry) const = 0;
```

Comment:
Much nicer signature🏅

Reply:
`const std::string_view`?

Reply:
I'm far from being a C++ expert, but doesn't mismatching const specifier between function declarations and definitions cause problems? Is this a standard thing to do, do all compilers support such mismatching use of consts?

So far we've 100% matched the consts between the definitions and declarations in new code AFAIK. That's a model I can understand, at least 😅 Not saying I'm necessarily against this, but this is something unfamiliar to me.

Reply:
Thanks for clarifying, that makes sense. I'm seeing now why is this preferable then, yeah. Cool trick 👍🏻 (and disregard my similar `const` comments)